### PR TITLE
Astro on npm

### PR DESCRIPTION
**Pull request by jansepar**
Reviewers: Anyone on Astro!
JIRA: https://mobify.atlassian.net/browse/HYB-646
Linked PRs: https://github.com/mobify/astro/pull/399

## Changes
- Brings the Astro SDK dependancy through npm rather than through git submodules

## How to test-drive this PR
- Checkout this branch, run `npm install`, make sure everything still works!
- Follow instructions in README `Developing against `develop` of astro` and ensure you can successfully develop against a local version of Astro.

## TODOS:
- [x] Upload the Astro folder to npm
- [x] Prompt the generator to show the license
- [x] Update the tutorial to use npm
- [x] Change works in both Android and iOS (SJ: I haven't fixed Android yet!)
- [x] CHANGELOG.md has been updated
- [x] Update the generator
- [x] Update the astro-sdk dependency to 0.7.0 (not -dev)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 To get up and running with scaffold, ensure the dependencies are installed:
 
-    npm run-script install-dependencies
+    npm install
```
Comment (jeremywiebe) [inline README.md]: @jansepar talked with Liz about this. Can we just have the generator as the last step? I guess we still need this documented but most people will use the generator and so we could automate that step also.

File: README.md [hunk 2]
```diff
@@ -22,22 +22,16 @@ Run the following command to open the iOS scaffold in Xcode:
...
 
-# Developing Astro from within your project
+    git clone git@github.com:mobify/astro.git
+    cd astro
+    npm link
 
-If you are making changes to Astro and testing them from within this project,
-`app/build-js.js` will detect if the `astro` submodule has been modified. If it has,
-it will automatically build `astro-client.js` and copy it into your project.
+Then run, `npm link astro-sdk` to use your locally developed version of Astro.
```
Comment (jeremywiebe) [inline README.md]: Should we clarify here that they need to return to their project directory before running `npm link astro-sdk`?

File: package.json [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
   "name": "astro-scaffold",
-  "version": "0.6.0",
+  "version": "0.7.0",
   "author": "Mobify",
-  "dependencies": {},
+  "dependencies": {
+    "bluebird": "~2.9.24"
+  },
+  "devDependencies": {
+    "astro-sdk": "^0.7.0-dev",
```
Comment (jeremywiebe) [inline package.json]: Shouldn't this be in the `dependencies` node?

File: ios/scaffold.xcworkspace/contents.xcworkspacedata [hunk 4]
```diff
@@ -2,10 +2,10 @@
 <Workspace
    version = "1.0">
    <FileRef
-      location = "group:../astro/ios/vendor/CordovaLib/CordovaLib.xcodeproj">
+      location = "group:../../astro/ios/vendor/CordovaLib/CordovaLib.xcodeproj">
```
Comment (jeremywiebe) [inline ios/scaffold.xcworkspace/contents.xcworkspacedata]: Shouldn't these point into `node_modules`?

File: README.md (hunk 1 above)
Comment (jansepar) [inline README.md]: The main issue is that you need to run `npm install` if you were to clone the project (so if you're not the person generating the project). We could put `npm install` in `build-js.sh`, but not sure if it's worth slowing things down for...

**Comment by kschmidtdev** [comment]
:+1:

**Pull request merged**